Result of `nixpkgs-review pr 175541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>nixui</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>colmena</li>
    <li>nix-doc</li>
    <li>nix-eval-jobs</li>
    <li>nix-plugins</li>
  </ul>
</details>
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>appvm</li>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>cached-nix-shell</li>
    <li>comma</li>
    <li>common-updater-scripts</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dep2nix</li>
    <li>dydisnix</li>
    <li>fusionInventory</li>
    <li>go2nix</li>
    <li>haskellPackages.cabal2nix-unstable</li>
    <li>haskellPackages.nix-paths</li>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.update-nix-fetchgit</li>
    <li>libnixxml</li>
    <li>lispPackages.quicklisp-to-nix</li>
    <li>lispPackages.quicklisp-to-nix-system-info</li>
    <li>nix (nixVersions.nix_2_9 ,nixVersions.stable)</li>
    <li>nix-bundle</li>
    <li>nix-direnv</li>
    <li>nix-du</li>
    <li>nix-index</li>
    <li>nix-pin</li>
    <li>nix-prefetch</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-cvs</li>
    <li>nix-prefetch-docker</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-hg</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-serve</li>
    <li>nix-simple-deploy</li>
    <li>nix-template</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nixStatic</li>
    <li>nixos-generators</li>
    <li>nixos-rebuild</li>
    <li>nixos-shell</li>
    <li>nixpkgs-review</li>
    <li>node2nix</li>
    <li>npins</li>
    <li>nuget-to-nix</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>python310Packages.nix-kernel</li>
    <li>python39Packages.nix-kernel</li>
    <li>rnix-lsp</li>
    <li>terranix</li>
    <li>tests.testers.nixosTest-example</li>
    <li>tests.trivial-builders.references</li>
    <li>update-nix-fetchgit</li>
    <li>vgo2nix</li>
    <li>vulnix</li>
    <li>wp4nix</li>
    <li>yarn2nix</li>
  </ul>
</details>

```
error: builder for '/nix/store/dlc0mk3knwaa432aqrr34zr73yshv4la-nix-plugins-9.0.0.drv' failed with exit code 2;
       last 10 log lines:
       >                  from /nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix/primops.hh:3,
       >                  from /build/source/extra-builtins.cc:2:
       > /nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix/eval-inline.hh: In instantiation of 'void nix::EvalState::forceValue(nix::Value&, Callable) [with Callable = nix::Pos]':
       > /build/source/extra-builtins.cc:56:25:   required from here
       > /nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix/eval-inline.hh:106:30: error: no match for call to '(nix::Pos) ()'
       >   106 |         throwEvalError(getPos(), "infinite recursion encountered");
       >       |                        ~~~~~~^~
       > make[2]: *** [CMakeFiles/nix-extra-builtins.dir/build.make:76: CMakeFiles/nix-extra-builtins.dir/extra-builtins.cc.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/nix-extra-builtins.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/dlc0mk3knwaa432aqrr34zr73yshv4la-nix-plugins-9.0.0.drv'.
error: builder for '/nix/store/rl8sc0r4gfxhfkm84ghccw53kp92ridi-nix-eval-jobs-0.0.6.drv' failed with exit code 1;
       last 10 log lines:
       > In file included from /nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix/eval.hh:3,
       >                  from ../src/nix-eval-jobs.cc:8:
       > /nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix/attr-set.hh:98:31: note: candidate: 'std::vector<const nix::Attr*> nix::Bindings::lexicographicOrder(const nix::SymbolTable&) const'
       >    98 |     std::vector<const Attr *> lexicographicOrder(const SymbolTable & symbols) const
       >       |                               ^~~~~~~~~~~~~~~~~~
       > /nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix/attr-set.hh:98:31: note:   candidate expects 1 argument, 0 provided
       > ../src/nix-eval-jobs.cc:297:66: error: no match for 'operator*' (operand type is 'nix::PosIdx')
       >   297 |                         recurse = state.forceBool(*attrv->value, *attrv->pos);
       >       |                                                                  ^~~~~~~~~~~
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/rl8sc0r4gfxhfkm84ghccw53kp92ridi-nix-eval-jobs-0.0.6.drv'.
error: 1 dependencies of derivation '/nix/store/s8s7lkfw3b7km23p46v82l92zn79ssgp-colmena-0.3.0.drv' failed to build
error: builder for '/nix/store/rjmb5cb1gdsvmsp00jkxirg9ybixpa87-nix-doc-0.5.3.drv' failed with exit code 101;
       last 10 log lines:
       >   exit status: 1
       >
       >   --- stderr
       >
       >
       >   error occurred: Command "/nix/store/r7r10qvsqlnvbzjkjinvscjlahqbxifl-gcc-wrapper-11.3.0/bin/c++" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-shared" "-Wall" "-Wextra" "-std=c++17" "-isystem" "/nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix" "-isystem" "/nix/store/xfia8xaq94slxmh5phqddqd8s1vcm7kg-boehm-gc-8.0.6-dev/include" "-L\"/nix/store/59galnfxqfih6146d1x32f3wdfv5cvxr-nix-2.9.0/lib\"" "-L\"/nix/store/xjiixs863ck3lkv1k1izw9rlipx6jscl-boehm-gc-8.0.6/lib\"" "-lnixexpr" "-lnixstore" "-lnixutil" "-lgc" "-lpthread" "-ldl" "-isystem" "/nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix" "-L\"/nix/store/59galnfxqfih6146d1x32f3wdfv5cvxr-nix-2.9.0/lib\"" "-lnixstore" "-lnixutil" "-isystem" "/nix/store/rslfmai8503xhyz8b5hb2qdxv2vgjfnb-nix-2.9.0-dev/include/nix" "-L\"/nix/store/59galnfxqfih6146d1x32f3wdfv5cvxr-nix-2.9.0/lib\"" "-lnixmain" "-DNIX_2_4_0" "-DNIX_2_6_0" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/nix-doc-plugin-a2f7562d43b72495/out/plugin.o" "-c" "plugin.cpp" with args "c++" did not execute successfully (status code exit status: 1).
       >
       >
       > warning: build failed, waiting for other jobs to finish...
       > error: build failed
       For full logs, run 'nix log /nix/store/rjmb5cb1gdsvmsp00jkxirg9ybixpa87-nix-doc-0.5.3.drv'.
       ```

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -88,6 +88,14 @@ in lib.makeExtensible (self: {
   nix_2_9 = common {
     version = "2.9.0";
     sha256 = "sha256-W6aTsTpCTb+vXQEXDjnKqetOuJmEfSuK2CXvAMqwo74=";
+    patches = [
+      # can be removed when updated to 2.9.1
+      (fetchpatch {
+        name = "fix-segfault-in-git-fetcher";
+        url = "https://github.com/NixOS/nix/commit/bc4759345538c89e1f045aaabcc0cafe4ecca12a.patch";
+        sha256 = "sha256-UrfH4M7a02yfE9X3tA1Pwhw4RacBW+rShYkl7ybG64I=";
+      })
+    ];
   };
 
   stable = self.nix_2_9;

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -90,7 +90,7 @@ in lib.makeExtensible (self: {
     sha256 = "sha256-W6aTsTpCTb+vXQEXDjnKqetOuJmEfSuK2CXvAMqwo74=";
   };
 
-  stable = self.nix_2_8;
+  stable = self.nix_2_9;
 
   unstable = lib.lowPrio (common rec {
     version = "2.8";

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -85,6 +85,11 @@ in lib.makeExtensible (self: {
     sha256 = "sha256-zldZ4SiwkISFXxrbY/UdwooIZ3Z/I6qKxtpc3zD0T/o=";
   };
 
+  nix_2_9 = common {
+    version = "2.9.0";
+    sha256 = "sha256-W6aTsTpCTb+vXQEXDjnKqetOuJmEfSuK2CXvAMqwo74=";
+  };
+
   stable = self.nix_2_8;
 
   unstable = lib.lowPrio (common rec {

